src/ts/content/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { PageContent } from "../common/types";
+import { buildSpecFrom } from "../parsers";
+import { OpenAPIObject } from "openapi3-ts";
+import debug from "../common/logging";
+
+const debugLog = debug("unmock:content:utils");
+
+let apiCheckResult: boolean;
```

Comment:
Perhaps this should be a mapping (`tabId/url -> boolean`), so we can cache multiple different tabs and not stick to the one caching result.

Reply:
The content script is run once for every page/tab so the same script should never have access to multiple DOMs.

Reply:
Ah, I thought it might be something like that. Good to know.